### DISCUSSION TITLE
Export to Image File

### DISCUSSION THREAD

**Discussion by sdgoldstein** [maxGraph/maxGraph#234] [Q&A]
I'd like to take a graph created with maxGraph and export it to an image file (format is not a concern).  Does anyone have sample code for this?  

Note that my graphs have overlay images.  Ideally, I'd like to simply remove those overlay images when exporting. 

Thanks for the help!

**Comment by tbouffard** [comment]
Hi, I have done this with mxGraph and start porting it to maxGraph.

I guess you to have an export feature in the application as maxGraph is used (otherwise, you can use external tools for that).
The solution will change whether you are using htmlLabels or not.

If `htmlLabels = false` (default), this is the easier part.
- SVG: in the DOM, take the content of the svg node within maxGraph container. Prepend it with the "svg" directive.
- PNG and JPG, you can use the generated SVG to fill img and HTML5 canvas and then produce the image.

If `htmlLabels = true` (which is my case, to be able to position the labels), this is more tricky
- SVG: you have to regenerate a dedicated SVG to have a fallback to SVG text for labels, as the maxGraph labels are using foreignObject+HTML and some editors don't support it (it should works if you intend to open the SVG file in a browser)
- PNG and JPG: same as in the other case.

The SVG generation I used is inspired from the implementation in draw.io:
- mxgraph implementation (provide links do the code in draw.io I used): https://github.com/process-analytics/bpmn-visualization-js/blob/v0.38.1/dev/ts/component/SvgExporter.ts
- maxGraph experimental implementation (not fully migrated): https://github.com/process-analytics/bpmn-visualization-js/blob/325e11930f6c14d73e2d8de8119c227e46f81294/dev/ts/component/SvgExporter.ts (taken from a POC https://github.com/process-analytics/bpmn-visualization-js/pull/2756)
- PNG and JPG generation and download: https://github.com/process-analytics/bpmn-visualization-js/blob/v0.38.1/dev/ts/component/download.ts
